tools/slims.py

```diff
@@ -44,71 +41,42 @@ def dna(self):
 
         return self._dna
 
-    @property
-    def fastq(self):
-        if not self.run_tag:
-            raise Exception('Can not fetch fastq without a set run tag.')
-
-        if not self._fastq:
-            records = slims.fetch('Content', conjunction()
-                                  .add(equals('cntn_id', self.sample_name))
-                                  .add(equals('cntn_fk_contentType', 22))
-                                  .add(equals('cntn_cstm_runTag', self.run_tag)))
-
-            if len(records) > 1:
-                raise Exception('More than 1 fastq somehow.')
-
-            if records:
-                self._fastq = records[0]
-
-        return self._fastq
 
     @property
     def fastqs(self):
         if not self._fastqs:
             records = slims.fetch('Content', conjunction()
-                                  .add(equals('cntn_id', self.sample_name))
+                                  .add(equals('cntn_id', self.content_id))
                                   .add(equals('cntn_fk_contentType', 22)))
             self._fastqs = records
 
         return self._fastqs
 
+
     @property
     def bioinformatics(self):
-        if not self.run_tag:
-            raise Exception('Can not fetch fastq without a set run tag.')
-
         if not self._bioinformatics:
             records = slims.fetch('Content', conjunction()
-                                  .add(equals('cntn_id', self.sample_name))
-                                  .add(equals('cntn_fk_contentType', 23))
-                                  .add(equals('cntn_cstm_runTag', self.run_tag)))
-
-            if len(records) > 1:
-                raise Exception('More than 1 bioinformatics somehow.')
-
-            if records:
-                self._bioinformatics = records[0]
+                                  .add(equals('cntn_id', self.content_id))
+                                  .add(equals('cntn_fk_contentType', 23)))
+            self._bioinformatics = records
 
         return self._bioinformatics
 
-    def add_bioinformatics(self, fields={}):
-        if self.bioinformatics:
-            raise Exception('Bioinformatics content already exists. Can not add.')
-        if not self.fastq:
+    def add_bioinformatics(self, originalContent, fields={}):
```

Comment:
If I understand correctly, the `fields` argument here allows the user to specify custom fields to be added. If that is the intention, does this function need better documentation to specify what fields can be specified? Maybe some exception handling is needed if invalid fields are specified (assuming this raises an exception in `slims.add`)?

Reply:
yes, good point. Also, making more verbose exception classes should definitely be something to do.